datarepo-clienttests/src/main/java/testscripts/RetrieveDRSSnapshot.java

```diff
@@ -0,0 +1,216 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import utils.DataRepoUtils;
+import utils.FileUtils;
+
+public class RetrieveDRSSnapshot extends runner.TestScript {
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public RetrieveDRSSnapshot() {
+    super();
+  }
+
+  private String datasetCreator;
+  private BillingProfileModel billingProfileModel;
+  private DatasetSummaryModel datasetSummaryModel;
+  private SnapshotSummaryModel snapshotSummaryModel;
+
+  private Storage storage;
+  private List<String> createdScratchFiles;
+  private String testConfigGetIngestbucket;
+
+  public void setup(Map<String, ApiClient> apiClients) throws Exception {
+    // pick the first user to be the dataset creator
+    List<String> apiClientList = new ArrayList<>(apiClients.keySet());
+    datasetCreator = apiClientList.get(0);
+
+    // get the ApiClient for the dataset creator --- and the snapshot creator?
+    ApiClient datasetCreatorClient = apiClients.get(datasetCreator);
+    ResourcesApi resourcesApi = new ResourcesApi(datasetCreatorClient);
+    RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+
+    storage = StorageOptions.getDefaultInstance().getService();
+    createdScratchFiles = new ArrayList<>();
+    testConfigGetIngestbucket = "jade-testdata"; // this could be put in DRUtils
+
+    // create a new profile
+    billingProfileModel =
+        DataRepoUtils.createProfile(resourcesApi, billingAccount, "profile-simple", true);
+
+    System.out.println("successfully created profile: " + billingProfileModel.getProfileName());
+
+    // make the create dataset request and wait for the job to finish
+    JobModel createDatasetJobResponse =
+        DataRepoUtils.createDataset(
+            repositoryApi, billingProfileModel.getId(), "dataset-simple.json", true);
+
+    // save a reference to the dataset summary model so we can delete it in cleanup()
+    datasetSummaryModel =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, createDatasetJobResponse, DatasetSummaryModel.class);
+
+    System.out.println("successfully created dataset: " + datasetSummaryModel.getName());
+
+    // load data into the new dataset
+    // note that there's a fileref in the dataset
+
+    // ingest a file
+    URI sourceUri = new URI("gs://jade-testdata/fileloadprofiletest/1KBfile.txt");
+    // todo generate a 1 byte file to load
+
+    String targetPath = "/testrunner/IngestFile/" + FileUtils.randomizeName("") + ".txt";
+
+    FileLoadModel fileLoadModel =
+        new FileLoadModel()
+            .sourcePath(sourceUri.toString())
+            .description("IngestFile")
+            .mimeType("text/plain")
+            .targetPath(targetPath)
+            .profileId(datasetSummaryModel.getDefaultProfileId());
+    JobModel ingestFileJobResponse =
+        repositoryApi.ingestFile(datasetSummaryModel.getId(), fileLoadModel);
```

Comment:
We are getting rid of the single file load end point. Please use the array form with an array of one